### PR TITLE
Feat/show modal after profile approved

### PR DESCRIPTION
**Pull request by filwas**

### CODE REVIEW

**Comment by filwas** [comment]
I believe there is an issue where the profile is being automatically verified whenever the user clicks "save and preview profile" in "edit profile". Due to this, my modal won't work when being redirected from the "edit profile" route.

File: apps/platform-app/prisma/migrations/20241105184615_add_approval_modal_shown/migration.sql
Comment (NerdbordMentor) [inline apps/platform-app/prisma/migrations/20241105184615_add_approval_modal_shown/migration.sql:2]: **suggestion**: Consider adding a comment explaining the purpose of the `wasApprovalModalShown` column for future reference.

File: apps/platform-app/src/app/[locale]/(profile)/(components)/TogglePublishButton/TogglePublishButton.tsx
Comment (NerdbordMentor) [inline apps/platform-app/src/app/[locale]/(profile)/(components)/TogglePublishButton/TogglePublishButton.tsx:13]: **suggestion**: Consider using a more descriptive name for the `onClose` function to clarify its purpose.

File: apps/platform-app/src/app/[locale]/(profile)/(components)/VerificationModal/VerificationModal.tsx
Comment (NerdbordMentor) [inline apps/platform-app/src/app/[locale]/(profile)/(components)/VerificationModal/VerificationModal.tsx:11]: **nitpick**: The `onClose` function type should be `() => void` instead of `() => {}` for better type clarity.

File: apps/platform-app/src/app/[locale]/(profile)/(components)/VerificationModal/VerificationModal.tsx
Comment (NerdbordMentor) [inline apps/platform-app/src/app/[locale]/(profile)/(components)/VerificationModal/VerificationModal.tsx:27]: **issue**: The condition should check if `profileStatus` is `PublishingState.APPROVED` instead of using `PublishingState.APPROVED` directly in the ternary operation.

File: apps/platform-app/src/app/[locale]/(profile)/(components)/VerificationModal/VerificationModal.tsx
Comment (NerdbordMentor) [inline apps/platform-app/src/app/[locale]/(profile)/(components)/VerificationModal/VerificationModal.tsx:43]: **issue**: The `bodyText` assignment should use `profileStatus === PublishingState.APPROVED` to determine the correct text.

**Review by NerdbordMentor — CHANGES_REQUESTED** [review]
## Overall Feedback

The solution demonstrates a good effort in implementing the required modals for profile approval and rejection. The code changes include the addition of a new column in the database, updates to the Prisma schema, and the implementation of a new `VerificationModal` component. However, there are some critical issues that need to be addressed before approval.

### Things Missing:

- [ ] The rejection modal does not include a specific rejection reason in the message, which is a requirement.

### Things That Went Well:

- The `VerificationModal` component is well-structured and integrates smoothly with the existing codebase.
- The styling for the modal is clean and consistent with the application's design.

**Review by Sumick — CHANGES_REQUESTED** [review]
Few improvements needed before mergin - we are on it during CC now

**Review by Sumick — COMMENTED** [review]
TODO after CC:

- [x] Add translations
- [x] Create `rejection-reason.model.ts`
- [ ] Create `findLatestRejectionForProfile.ts` that will return that model
- [ ] Create rejection.service or extend it if one exists to it will return latest rejection for profile
- [ ] Fetch latest rejection for profile if profile is rejected to show why profile was rejected
- [ ] Save & preview should not run verification agent but make profile as draft - then Publish button should appear

File: apps/platform-app/src/app/[locale]/(profile)/(components)/TogglePublishButton/TogglePublishButton.tsx
Comment (NerdbordMentor) [inline apps/platform-app/src/app/[locale]/(profile)/(components)/TogglePublishButton/TogglePublishButton.tsx:19]: **issue**: The `TogglePublishButton` component should not be async as it doesn't perform any asynchronous operations directly.

File: apps/platform-app/src/app/[locale]/(profile)/(components)/TogglePublishButton/TogglePublishButton.tsx
Comment (NerdbordMentor) [inline apps/platform-app/src/app/[locale]/(profile)/(components)/TogglePublishButton/TogglePublishButton.tsx:24]: **suggestion**: Add `state` as a dependency in the `useEffect` dependency array to ensure it runs when `state` changes.

File: apps/platform-app/src/app/[locale]/(profile)/(components)/VerificationModal/VerificationModal.tsx
Comment (NerdbordMentor) [inline apps/platform-app/src/app/[locale]/(profile)/(components)/VerificationModal/VerificationModal.tsx:38]: **suggestion**: Consider using a more descriptive title for the Accordion, such as "Rejection Details".

File: apps/platform-app/src/app/[locale]/(profile)/_actions/queries/findLatestRejectionReason.ts
Comment (NerdbordMentor) [inline apps/platform-app/src/app/[locale]/(profile)/_actions/queries/findLatestRejectionReason.ts:7]: **suggestion**: Consider adding a return type to the `findLatestRejectionReason` function for better type safety.

File: apps/platform-app/src/app/[locale]/(profile)/_models/rejection-reason.model.ts
Comment (NerdbordMentor) [inline apps/platform-app/src/app/[locale]/(profile)/_models/rejection-reason.model.ts:7]: **suggestion**: Add a return type to `createRejectionReasonModel` for better type safety.

**Review by NerdbordMentor — APPROVED** [review]
The solution provided by the author, filwas, meets the task requirements by implementing two modals to provide feedback to the user about their profile approval or rejection. The code changes include the addition of a new "VerificationModal" component, updates to the language JSON files to include modal text, and modifications to the Prisma schema and various components to support the new functionality. The rejection modal correctly includes a reason for rejection, fulfilling the task requirements.

### Things Missing: 

- [] No critical issues or missing features were identified that prevent approval.

### Things That Went Well:

- The implementation of the "VerificationModal" component is well-structured and integrates smoothly with the existing codebase.
- The use of i18n for modal text ensures that the application remains accessible to users in different languages.
- The addition of the "wasApprovalModalShown" field in the Prisma schema is a thoughtful touch to track user interactions.
- The code changes are comprehensive and cover all necessary aspects to meet the task requirements.

File: apps/platform-app/src/backend/profile/rejection.service.ts
Comment (Sumick) [inline apps/platform-app/src/backend/profile/rejection.service.ts:21]: **nit:** getLatestRejectionReason

**Review by Sumick — APPROVED** [review]
Looks good, let's just do some cleanup before merging

**Review by transcendence12 — APPROVED** [review]

**Review by Wiecek-K — APPROVED** [review]
lgtm

**Review by Wiecek-K — CHANGES_REQUESTED** [review]
Change the branch you are merging into!

File: apps/platform-app/src/app/[locale]/(profile)/(components)/MyProfile/ProfileTopBar/ProfileTopBar.tsx
Comment (meczajkowski) [inline apps/platform-app/src/app/[locale]/(profile)/(components)/MyProfile/ProfileTopBar/ProfileTopBar.tsx:72]: I would change it to null if does not exist

**Review by meczajkowski — APPROVED** [review]
Great job!

Please run `npm run format` before merging to `staging` if PR #360 would be merged first.

**Review by Wiecek-K — APPROVED** [review]

File: apps/platform-app/src/app/[locale]/(profile)/(components)/TogglePublishButton/TogglePublishButton.tsx
Comment (Sumick) [inline apps/platform-app/src/app/[locale]/(profile)/(components)/TogglePublishButton/TogglePublishButton.tsx:66]: **nit:** Why not just `showPopup`?

File: apps/platform-app/src/app/[locale]/(profile)/_actions/mutations/saveMyProfile.ts
Comment (Sumick) [inline apps/platform-app/src/app/[locale]/(profile)/_actions/mutations/saveMyProfile.ts:128]: **issue:** It will never work, because you hardcoded state to DRAFT, so it will never be PENDING to meet this condition.

```ts
const updatedState = hasCommonFields(changedFields, profilePendingFields)
    ? PublishingState.PENDING
    ? PublishingState.DRAFT
    : payload.state
```

**Review by Sumick — CHANGES_REQUESTED** [review]
I found one blocking bug and minor nitpick

File: apps/platform-app/src/app/[locale]/(profile)/(components)/TogglePublishButton/TogglePublishButton.tsx
Comment (filwas) [reply apps/platform-app/src/app/[locale]/(profile)/(components)/TogglePublishButton/TogglePublishButton.tsx:66]: Maybe this was only due to the fact that both the app and db were running locally, but there were situations where I had showPopup as true, while state was DRAFT or PENDING.

File: apps/platform-app/src/app/[locale]/(profile)/_actions/mutations/saveMyProfile.ts
Comment (filwas) [reply apps/platform-app/src/app/[locale]/(profile)/_actions/mutations/saveMyProfile.ts:127]: This is indeed how I intended o leave this. The task was to make the "save and preview" button in Edit Profile NOT fire `runEvaluateProfileAgent()`. I could delete this whole block, but decided to leave it if we want to add "save and publish" button in the future.

File: apps/platform-app/src/app/[locale]/(profile)/(components)/MyProfile/ProfileDetails/ProfileDetails.tsx
Comment (Sumick) [inline apps/platform-app/src/app/[locale]/(profile)/(components)/MyProfile/ProfileDetails/ProfileDetails.tsx:15]: **nit:** comment to remove

**Review by Sumick — APPROVED** [review]
Looks good! Let's just remove commented dead code before merging

**Pull request merged**
